src/uwtools/resources/jsonschema/upp.jsonschema

```diff
@@ -0,0 +1,188 @@
+{
+  "properties": {
+    "upp": {
+      "additionalProperties": false,
+      "properties": {
+        "execution": {
+          "$ref": "urn:uwtools:execution"
+        },
+        "files_to_copy": {
+          "$ref": "urn:uwtools:files-to-stage"
+        },
+        "files_to_link": {
+          "$ref": "urn:uwtools:files-to-stage"
+        },
+        "namelist": {
+          "additionalProperties": false,
+          "anyOf": [
+            {
+              "required": [
+                "base_file"
+              ]
+            },
+            {
+              "required": [
+                "update_values"
+              ]
+            }
+          ],
+          "properties": {
+            "base_file": {
+              "type": "string"
+            },
+            "update_values": {
+              "additionalProperties": false,
+              "properties": {
+                "model_inputs": {
+                  "additionalProperties": false,
+                  "properties": {
+                    "datestr": {
+                      "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{2}:[0-9]{2}:[0-9]{2}$",
+                      "type": "string"
+                    },
+                    "filename": {
+                      "maxLength": 256,
+                      "type": "string"
+                    },
+                    "filenameflat": {
+                      "maxLength": 256,
+                      "type": "string"
+                    },
+                    "filenameflux": {
+                      "maxLength": 256,
+                      "type": "string"
+                    },
+                    "grib": {
+                      "enum": [
+                        "grib2"
+                      ],
+                      "type": "string"
+                    },
+                    "ioform": {
+                      "enum": [
+                        "binarynemsio",
+                        "netcdf"
+                      ],
+                      "type": "string"
+                    },
+                    "modelname": {
+                      "enum": [
+                        "FV3R",
+                        "3DRTMA",
+                        "GFS",
+                        "RAPR",
+                        "NMM"
+                      ],
+                      "type": "string"
+                    }
+                  },
+                  "type": "object"
+                },
+                "nampgb": {
```

Comment:
I don't think so, because the user might provide a mostly-correct `base_file` and only want to update a subset of the values. I think we can only describe what keys they _may_ provide and what their values look like; I don't think we can say that the _must_ provide anything. Please let me know if you see it otherwise.